docker/Dockerfile

```diff
@@ -0,0 +1,58 @@
+FROM gcc:11 as build
+
+WORKDIR /build
```

Comment:
One additional thought: Could you try to use different words for container stage, workdir and cmake build dir? At the moment, all three are `build` and it's a bit tricky to keep track of which is which sometimes.